pkg/i2gw/provider.go

```diff
@@ -99,8 +101,90 @@ type GatewayResources struct {
 	UDPRoutes  map[types.NamespacedName]gatewayv1alpha2.UDPRoute
 
 	ReferenceGrants map[types.NamespacedName]gatewayv1beta1.ReferenceGrant
+
+	GatewayExtensions map[types.NamespacedName]unstructured.Unstructured
```

Comment:
why is this field needed here

Reply:
This was an alternative that I'd discussed earlier with @sawsa307 - instead of modifying the existing path at all, we'd just add `GatewayExtensions` type that output providers could optionally read from. I slightly prefer that over the `GatewayContext` struct as it's less disruptive and easy to ignore for implementations that don't support extensions.

Imagine that instead of `unstructured.Unstructured` here we had a `GatewayExtension` struct that had all the content of `GatewayContext` except for the actual Gateway. I can be ok with the `GatewayContext` approach as well, I just generally preferred the optionality of this kind of approach.

Reply:
Yes, as Rob mentioned, the advantage of doing this as a map is, we can determine if we need to print `GatewayExtensions` based on if the map is empty.

The reason I used `unstructured.Unstructured` here is so we can hold Gateway Extensions for all providers. If we want to have a strongly typed field, it would have to be a field for each provider, and iterate over the supported Gateway extensions, something like:

```
type GatewayResources struct {
   ...
   
   Gce GceGatewayExtensions
}

type GceGatewayExtensions struct {
   GCPBackendPolicies map[types.NamespacedName]gkegatewayv1.GCPBackendPolicies
   GCPGatewayPolicies map[types.NamespacedName]gkegatewayv1.GCPGatewayPolicy
   HealthCheckPolicies map[types.NamespacedName]gkegatewayv1.HealthCheckPolicy
}
```

When printing results, we will also need to iterate over the list of extensions. This is my interpretation as an alternative for strongly typed, and I think it could be a valid option as well.